src/service/scenarios/delete_scenarios_service.py

```diff
@@ -0,0 +1,237 @@
+                    company_id,
+                    scenario["metric"],
+                    scenario["value"],
+                )
+                if scenario_is_deleted:
+                    scenarios_deleted += 1
+        else:
+            for scenario in scenarios:
+                scenario_is_deleted = self.delete_scenario_from_details(
+                    scenario["scenario_id"], scenario["metric_id"]
+                )
+                if scenario_is_deleted:
+                    scenarios_deleted += 1
+
+        return scenarios_deleted
+
+    def delete_currency(self, metric_id: str) -> bool:
+        try:
+            query = """
+            DELETE FROM {table_name} WHERE metric_id = '{metric_id}'
+            """.format(
+                table_name="currency_metric", metric_id=metric_id
+            )
+
+            self.session.execute(query)
+            self.session.commit()
+            return True
+        except Exception as error:
+            self.logger.info(error)
+            return False
+
+    def delete_scenario_metric(self, scenario_id: str, metric_id: str) -> bool:
+        currency_is_deleted = self.delete_currency(metric_id)
+
+        if currency_is_deleted:
+            try:
+                query = """
+                    DELETE FROM {table_name}
+                    WHERE scenario_id = '{scenario_id}'
+                    AND metric_id = '{metric_id}'
+                """.format(
+                    table_name="scenario_metric",
+                    scenario_id=scenario_id,
+                    metric_id=metric_id,
+                )
+
+                self.session.execute(query)
+                self.session.commit()
+                return True
+            except Exception as error:
+                self.logger.info(error)
+                return False
+        else:
+            return False
+
+    def delete_metric(self, scenario_id: str, metric_id: str) -> bool:
+        scenario_metric_is_deleted = self.delete_scenario_metric(scenario_id, metric_id)
+        if scenario_metric_is_deleted:
+            try:
+                query = """
+                DELETE FROM {table_name} WHERE id = '{metric_id}'
+                """.format(
+                    table_name="metric", metric_id=metric_id
+                )
+
+                self.session.execute(query)
+                self.session.commit()
+                return True
+            except Exception as error:
+                self.logger.info(error)
+                return False
+        else:
+            return False
+
+    def delete_scenario_from_details(self, scenario_id: str, metric_id) -> bool:
+        metric_is_deleted = self.delete_metric(scenario_id, metric_id)
+        if metric_is_deleted:
+            is_scenario = self.scenario_is_in_scenario_metric(scenario_id)
+            if not is_scenario:
+                period_id = self.get_time_period_of_scenario(scenario_id)
+                scenario_deleted = False
+                try:
```

Comment:
I don't think we should delete each record validating that the other one was deleted, we should delete all records in the same query, I mean with the same query delete scenario, time period, metric, scenario metric, and currency metric for each scenario that we have, you can avoid the use of if else.
And if you use transactions when something fails, we'll have all records and we can try again.